### PR TITLE
[chip-tool] Get logging code to have a dedicated non-generated class that includes generated code

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

Commands/Attributes/Events logging code in `chip-tool`  is a mix of generated and non-generated code but all of it lives into `commands.zapt`, which makes it harder to maintain than necessary.

#### Change overview
 * Move non-generated parts of the logging code to a dedicated class
 * Move the generated parts to a dedicated template
 * Use the result of the dedicated template into the non-generated class

#### Testing
I have manually checked that the logging code still works.

### CODE REVIEW

File: examples/chip-tool/commands/clusters/DataModelLogger.h [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    {
+        char buffer[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
+        if (CHIP_NO_ERROR ==
+            chip::Encoding::BytesToUppercaseHexString(value.data(), value.size(), &buffer[0], CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE))
+        {
+            DataModelLogger::LogString(label, indent, buffer);
+        }
+        else
+        {
+            DataModelLogger::LogString(label, indent, std::to_string(value.size()));
```
Comment (tcarmelveilleux) [inline examples/chip-tool/commands/clusters/DataModelLogger.h:51]: Can you improve this to say "Elided value too large of  size N" ?

File: examples/chip-tool/templates/logging/DataModelLogger-src.zapt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{{/unless}}
+{{/zcl_structs}}
+{{/zcl_clusters}}
+
+{{#zcl_clusters}}
+{{#zcl_events}}
+CHIP_ERROR DataModelLogger::LogValue(const char * label, size_t indent, const chip::app::Clusters::{{asUpperCamelCase parent.name}}::Events::{{asUpperCamelCase name}}::DecodableType & value)
+{
+  DataModelLogger::LogString(label, indent, "{");
+{{#zcl_event_fields}}
```
Comment (tcarmelveilleux) [inline examples/chip-tool/templates/logging/DataModelLogger-src.zapt:24]: Is this going to be generated for embedded platforms as well?

Comment (vivien-apple) [reply examples/chip-tool/templates/logging/DataModelLogger-src.zapt:24]: Yes. It is already.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]
Fast tracking, given tool changes.

**Pull request merged**